indico/web/client/js/react/components/principals/PrincipalField.jsx

```diff
@@ -0,0 +1,137 @@
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+import principalsURL from 'indico-url:core.principals';
+
+import React, {useEffect, useState} from 'react';
+import PropTypes from 'prop-types';
+import {Icon, List} from 'semantic-ui-react';
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
+import {camelizeKeys} from 'indico/utils/case';
+import {UserSearch} from './Search';
+import {EmptyPrincipalListItem, PendingPrincipalListItem, PrincipalListItem} from './items';
+
+import './items.module.scss';
+
+
+/**
+ * A field that lets the user select a user.
+ */
+const PrincipalField = (props) => {
+    const {value, disabled, onChange, onFocus, onBlur, favoriteUsersController} = props;
+    const [favoriteUsers, [handleAddFavorite, handleDelFavorite]] = favoriteUsersController;
+
+    const [details, setDetails] = useState(null);
+
+    const markTouched = () => {
+        onFocus();
+        onBlur();
+    };
+
+    useEffect(() => {
+        if (!value) {
+            if (details) {
+                // clear old details
+                setDetails(null);
+            }
+            return;
+        }
+        if (details && details.identifier === value) {
+            // nothing to do
+            return;
+        }
+        const source = indicoAxios.CancelToken.source();
+        (async () => {
+            let response;
+            try {
+                response = await indicoAxios.post(principalsURL(), {values: [value]}, {cancelToken: source.token});
+            } catch (error) {
+                handleAxiosError(error);
+                return;
+            }
+            setDetails(camelizeKeys(response.data[value]));
+        })();
+
+        return () => {
+            source.cancel();
+        };
+    }, [details, value]);
+
+    const handleAddItem = principal => {
+        onChange(principal.identifier);
+        markTouched();
+    };
+    const handleClear = () => {
+        onChange(null);
+        markTouched();
+    };
+
+    const searchTrigger = handleClick => (
+        <Icon styleName="button" name="search" size="large" disabled={disabled} onClick={handleClick} />
+    );
+    const userSearch = (
+        <UserSearch triggerFactory={searchTrigger}
+                    existing={value ? [value] : []}
+                    onAddItems={handleAddItem}
+                    onClose={markTouched}
+                    favorites={favoriteUsers}
+                    disabled={disabled}
+                    single />
+    );
+
+    return (
+        <div className="ui input">
+            <div className="fake-input">
```

Comment:
FYI this class is defined in indico-sui-theme since it needs access to the SUI variables. It applies the same styling any other `<input>` would get.